metric/system/process/helpers_windows.go

```diff
@@ -35,3 +38,23 @@ func isNonFatal(err error) bool {
 		errors.Is(err, syscall.EINVAL) ||
 		errors.Is(err, windows.ERROR_INVALID_PARAMETER) || errors.Is(err, NonFatalErr{})
 }
+
+func processesToIgnore() (m map[uint64]struct{}) {
+	// processesToIgnore checks if we should ignore the pid, to avoid elevated permissions
+
+	// LSASS.exe is a process which has no useful cmdline arguments, we should ignore acessing such process to avoid triggering Windows ASR rules
+	// we can query pid for LASASS.exe from registry
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, "SYSTEM\\CurrentControlSet\\Control\\Lsa", registry.READ)
+	if err != nil {
```

Comment:
Under what circumstances do we get an error here?

Reply:
If you're running the elastic-agent in unprivileged mode and the `elastic-agent-user` doesn't have enough permissions to read the registry.

Reply:
or if the key is not present (which is highly unlikely in this scenario)

Reply:
In that case we just emit a warning and fall back to previous behaviour though, so no risk of breaking a currently working installation via this change.

Reply:
@swiatekm that's what we do. See here https://github.com/elastic/elastic-agent-system-metrics/blob/33fcaec96311c1b468e7fb02284b71740b5257e9/metric/system/process/helpers_windows.go#L48-L51

Just log the warning and return.